frontend/src/routes/Applications/CreateApplication/CreateApplicationArgo.tsx

```diff
@@ -92,10 +92,30 @@ export function CreateApplicationArgo() {
 
   const { cancelForm, submitForm } = useContext(LostChangesContext)
   const [createdResource, setCreatedResource] = useState<any>()
+  const [applicationSets, setApplicationSets] = useState<ApplicationSet[]>()
+  const [loadingAppSets, setLoadingAppSets] = useState(true)
+
+  // instead of burdoning recoil with appsets, use old fashioned fetch
+  // opening wizard may take longer, but argo wizards are probably seldom used
```

Comment:
Looking at how the application sets are used by the wizard, it seems the only purpose is to pre-populate the Git and Helm URLs in the dropdowns. In an environment with thousands of application sets, I think it is likely that there will be far fewer unique sources, so it would be more efficient to calculate the unique repoURLs from the cached data on the backend, and then just send that information to the UI.

This is the sort of thing I had hoped would be investigated as part of the spike [ACM-17870](https://issues.redhat.com/browse/ACM-17870), which was closed without any findings.

Reply:
Yes--it's likely there are fewer unique sources--but I think it's even likelier that if there are thousands of appsets that they're being created by the cli and not this wizard, and that my awesome work adding a new backend api would go unappreciated :(

Reply:
OK, so we'll consider this a separate problem.